### ISSUE TITLE
Clarify Scala version compatibility for latest snapshots

### ISSUE THREAD

**Issue by boltzmann-brain** [microsoft/SynapseML#1188]
**Describe the bug**

I've been trying to use the latest mmlspark.explainers module. I can't run the [TabularSHAP notebook demo](https://github.com/microsoft/SynapseML/blob/master/notebooks/Interpretability%20-%20Tabular%20SHAP%20explainer.ipynb) on Databricks because of what seems like Scala version incompatibilities. It would be great if you could clarify which versions of Scala we would need to use that module. I'm using the following snapshot:
1.0.0-rc3-148-87ec5f74-SNAPSHOT

**To Reproduce**

On a Databricks cluster with Spark 3.0.1 and Scala 2.12, the notebook linked above fails at the following line:

```
shap = TabularSHAP(
    inputCols=categorical_features + numeric_features,
    outputCol="shapValues",
    numSamples=5000,
    model=model,
    targetCol="probability",
    targetClasses=[1],
    backgroundData=training.orderBy(rand()).limit(100).cache(),
)
```
with the error `java.lang.NoClassDefFoundError: scala/collection/IterableOnce`

It seems that this `IterableOnce` class is available only in scala-library 2.13. If this is correct does that mean I can't use it on Databricks with Scala 2.12?

When I add the jars for Scala-library 2.13 and scalatest 2.13 to my cluster, that error goes away, but I get another error that, according to [posts like this](https://stackoverflow.com/questions/60466630/databricks-error-java-lang-nosuchmethoderror-scala-predef-refarrayopsljava-l), seems related to inappropriate 'mixing' of Scala versions:
`java.lang.NoSuchMethodError: scala.Predef$.refArrayOps([Ljava/lang/Object;)`

**Expected behavior**

Without indications otherwise, I had expected the notebook to work with the specified requirements for the official release of Spark 2.4 and Scala 2.11. The notebook should also note that it's for the snapshot version, not the official release.

**Info (please complete the following information):**
 - MMLSpark Version: 1.0.0-rc3-148-87ec5f74-SNAPSHOT
 - Spark Version 3.0.1
 - Spark Platform Databricks
 - Scala 2.12

@memoryz is the code owner I believe

**Comment by AllardJM** [comment]
I have wondered the same for Scala and spark itself. I am on an older version of spark and i wonder if errors I see are related to that. The documentation here is lacking.

**Comment by memoryz** [comment]
The latest mmlspark.explainers module should work on Spark 3+ (including Spark 3.0.1), and Scala 2.12.

I just tested the [TabularSHAP notebook](https://github.com/microsoft/SynapseML/blob/master/notebooks/Interpretability%20-%20Tabular%20SHAP%20explainer.ipynb) on **Azure Databricks runtime 7.3 LTS** (Apache Spark 3.0.1, Scala 2.12), and the notebook runs successfully.

I used the same mmlspark version as you mentioned:
![image](https://user-images.githubusercontent.com/6992432/133555353-5a06560e-d623-4b91-aba4-3edd92462882.png)

Adding Scala-library 2.13 won't work as our bits are compiled against Scala 2.12.

Any chance you're running this from Databricks on AWS? We've tested this in Azure Databricks and Azure Synapse, but not on AWS Databricks.

Could you please paste the full error stack for this error `java.lang.NoClassDefFoundError: scala/collection/IterableOnce`?

**Comment by boltzmann-brain** [comment]
Apologies---the error was caused by the fact that I had the wrong version of spray-json installed. The version I installed was for Scala 2.13. Once I installed the spray-json for Scala 2.12, the error went away and I was able to run the entire notebook.

And yes, I was using DataBricks AWS, so I can confirm that it works on Databricks AWS, runtime 7.3 LTS (Apache Spark 3.0.1, Scala 2.12). Thanks!
